### PR TITLE
[Compute] `az ppg create/update`: Add parameter `--intentvmsizes` to support one or more VM sizes and `--zone` to support specifying availability zone where the ppg should be created

### LINKED ISSUE #22466: az ppg create and az ppg update commands to support User Intent
**Related command**
Optional parameters and values need to be added to the following commands to support user intent:
- az ppg create
- az ppg update

**Is your feature request related to a problem? Please describe.**
No. This is a new feature request.

**Describe the solution you'd like**
Intent
Intent is an optional parameter which you can provide while creating a [proximity placement group](https://docs.microsoft.com/en-us/azure/virtual-machines/co-location). It can also be added or modified while updating the PPG after deallocating all VMs in the proximity placement group.

The intent allows you to provide one or more [VM Sizes](https://docs.microsoft.com/en-us/azure/virtual-machines/sizes) that you require to be deployed as part of the proximity placement group(across single VMs, Availability Sets and VM Scale Sets).

Availability zone
In addition to the intent, an optional availability zone can be provided using zones parameter in case the proximity placement group must be created within a specific zone. Please note:

The availability zone can be provided only during proximity placement group creation and cannot be modified subsequently; unlike the intent, it cannot be updated even when the proximity placement group has all VM resources deallocated

The availability zone can be provided only when an intent is provided

Only one zone can be indicated as part of the availability zone

Please note that specifying a zone does not allow the proximity placement group to span the entire zone. It only helps select a data center within the specified zone rather than anywhere in the region.

The PPG create and update API with zones and intent parameters will look like the following:

API Version
API version must be 2021-11-01 or higher for using proximity placement group Intent

{
"apiVersion": "2021-11-01", 
"type": "Microsoft.Compute/proximityPlacementGroups", 
"name": "[parameters('ppgName')]", 
"location": "[resourceGroup().location]", 
“zones”: [“2”],
"properties": { 
    "proximityPlacementGroupType": "Standard", 
    “intent”: {
     "vmSizes": ["Standard_E64s_v4", "Standard_M416ms_v2"]
  }
}

Example of az pgg create command with user intent: 
az ppg create --name MyProximityPlacementGroup --resource-group MyResourceGroup --location eastus2 --type 'Standard' --zone 1 --intentvmsizes 'Standard_E64s_v4, Standard_M416ms_v2' 

| New Parameter     | Description/Value |
| ----------- | ----------- |
| zone | Availability Zone where the PPG should be created - accepted values 1,2,3 |
| intentvmsizes | One or more [VM Sizes](https://docs.microsoft.com/en-us/azure/virtual-machines/sizes) that you require to be deployed as part of the PPG (across single VMs, Availability Sets and VM Scale Sets) - it is an array of strings |

Note that zone can only be provided when intentvmsizes is provided; otherwise error should be thrown

Example of az ppg update with user intent:
az ppg create --name MyProximityPlacementGroup --resource-group MyResourceGroup --intentvmsizes 'Standard_E64s_v4, Standard_M416ms_v2' 

| New Parameter     | Description/Value |
| ----------- | ----------- |
| intentvmsizes | One or more [VM Sizes](https://docs.microsoft.com/en-us/azure/virtual-machines/sizes) that you require to be deployed as part of the PPG (across single VMs, Availability Sets and VM Scale Sets) - it is an array of strings |

Note that zone is NOT a parameter to be added in the update command as it can be used only during PPG creation time

**Describe alternatives you've considered**
It is a new feature request - so no new alternatives

**Additional context**
| Role     | Contact |
| ----------- | ----------- |
| Developer | aspand@microsoft.com |
| PM | dejv@microsoft.com |
| Reviewer | abver@microsoft.com |

Link to the OpenAPI (swagger) spec: https://github.com/Azure/azure-rest-api-specs/pull/18159

### PR DESCRIPTION
**Pull request by yanzhudd**
**Description**

Add parameter `--intentvmsizes` in `az ppg create` to support one or more vm sizes that you require to be deployed as part of the ppg and `--zone` to support specifying availability zone where the ppg should be created.
Add parameter `--intentvmsizes` in `az ppg update` to support one or more vm sizes that you require to be deployed as part of the ppg.
Close #22466

**Testing Guide**

**History Notes**

[Compute] `az ppg create/update`: Add parameter `--intentvmsizes` to specify possible sizes of VM that can be created in the proximity placement group
[Compute] `az ppg create`: Add parameter `--zone` to support specifying availability zone where the ppg should be created

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Compute

File: src/azure-cli/azure/cli/command_modules/vm/_help.py [hunk 1]
```diff
@@ -629,6 +629,14 @@
...
         az ppg create --name MyProximityPlacementGroup --resource-group MyResourceGroup
     crafted: true
+  - name: Create a proximity placement group with specifying vm sizes that can be created.
+    text: |
+        az ppg create --name MyProximityPlacementGroup --resource-group MyResourceGroup \\
+            --intentvmsizes Standard_E64s_v4 Standard_M416ms_v2
+  - name: Create a proximity placement group with specifying vm sizes that can be created and availability zone.
+    text: |
+        az ppg create --name MyProximityPlacementGroup --resource-group MyResourceGroup \\
+            --intentvmsizes Standard_E64s_v4 Standard_M416ms_v2 --zone 1
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/vm/_help.py:639]:
```suggestion
  - name: Create a proximity placement group with specifying VM sizes that can be created.
    text: |
        az ppg create --name MyProximityPlacementGroup --resource-group MyResourceGroup \\
            --intentvmsizes Standard_E64s_v4 Standard_M416ms_v2
  - name: Create a proximity placement group with specifying VM sizes that can be created and availability zone.
    text: |
        az ppg create --name MyProximityPlacementGroup --resource-group MyResourceGroup \\
            --intentvmsizes Standard_E64s_v4 Standard_M416ms_v2 --zone 1
```

File: src/azure-cli/azure/cli/command_modules/vm/_help.py [hunk 2]
```diff
@@ -629,6 +629,14 @@
     text: |
         az ppg create --name MyProximityPlacementGroup --resource-group MyResourceGroup
     crafted: true
+  - name: Create a proximity placement group with specifying vm sizes that can be created.
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/vm/_help.py:632]:
```suggestion
  - name: Create a proximity placement group with specifying VM sizes that can be created.
```

File: src/azure-cli/azure/cli/command_modules/vm/_help.py [hunk 3]
```diff
@@ -629,6 +629,14 @@
...
+    text: |
+        az ppg create --name MyProximityPlacementGroup --resource-group MyResourceGroup \\
+            --intentvmsizes Standard_E64s_v4 Standard_M416ms_v2
+  - name: Create a proximity placement group with specifying vm sizes that can be created and availability zone.
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/vm/_help.py:636]:
```suggestion
  - name: Create a proximity placement group with specifying VM sizes that can be created and availability zone.
```

File: src/azure-cli/azure/cli/command_modules/vm/_help.py [hunk 4]
```diff
@@ -654,6 +662,11 @@
 helps['ppg update'] = """
 type: command
 short-summary: Update a proximity placement group
+examples:
+  - name: Update a proximity placement group with specifying vm sizes that can be created.
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/vm/_help.py:666]:
```suggestion
  - name: Update a proximity placement group with specifying VM sizes that can be created.
```

File: src/azure-cli/azure/cli/command_modules/vm/_params.py [hunk 5]
```diff
@@ -1412,6 +1412,13 @@ def load_arguments(self, _):
         c.argument('ppg_type', options_list=['--type', '-t'], help="The type of the proximity placement group. Allowed values: Standard.")
         c.argument('tags', tags_type)
 
+    with self.argument_context('ppg create', min_api='2021-11-01') as c:
+        c.argument('zone', nargs='*', help="Specifies the availability zone where virtual machine, virtual machine scale set or availability set associated with the proximity placement group can be created. Accepted values 1,2,3.")
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/vm/_params.py:1416]: Could we reuse the `zone_type`
https://github.com/Azure/azure-cli/blob/f7133cb5e2af91ee2cd8673d86885cf292294cd9/src/azure-cli-core/azure/cli/core/commands/parameters.py#L306-L311

Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/vm/_params.py:1416]: Please use the first person voice in the help message
```suggestion
        c.argument('zone', nargs='*', help="Specify the availability zone where virtual machine, virtual machine scale set or availability set associated with the proximity placement group can be created. Accepted values 1,2,3.")
```

File: src/azure-cli/azure/cli/command_modules/vm/_params.py [hunk 6]
```diff
@@ -1412,6 +1412,13 @@ def load_arguments(self, _):
...
+
+    for scope in ['ppg create', 'ppg update']:
+        with self.argument_context(scope, min_api='2021-11-01') as c:
+            c.argument('intentvmsizes', nargs='*', help="Specifies possible sizes of virtual machines that can be created in the proximity placement group.")
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/vm/_params.py:1420]: It is suggested to mark the configuration of `min_api` at a finer level
```suggestion
        with self.argument_context(scope) as c:
            c.argument('intentvmsizes', nargs='*', min_api='2021-11-01', help="Specifies possible sizes of virtual machines that can be created in the proximity placement group.")
```

File: src/azure-cli/azure/cli/command_modules/vm/_validators.py [hunk 7]
```diff
@@ -2129,6 +2129,16 @@ def _disk_encryption_set_format(cmd, namespace, name):
 # endregion
 
 
+def process_ppg_create_namespace(namespace):
+    """
+    The availability zone can be provided only when an intent is provided
+    """
+    if namespace.zone and not namespace.intentvmsizes:
+        raise RequiredArgumentMissingError('Please use parameter --intentvmsizes to specify possible sizes of virtual '
+                                           'machines that can be created in the proximity placement group.')
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/vm/_validators.py:2138]: > The availability zone can be provided only when an intent is provided

I suggest reflecting the relationship between `--intentvmsizes` and `--zone` in the error message

**Comment by zhoxing-ms** [comment]
When the content of this PR needs to be written into the history notes is more than one, please split it into multiple points in the PR description, such as
Screenshot 2022-07-11 173839

File: src/azure-cli/azure/cli/command_modules/vm/_params.py [hunk 8]
```diff
@@ -1412,6 +1412,13 @@ def load_arguments(self, _):
         c.argument('ppg_type', options_list=['--type', '-t'], help="The type of the proximity placement group. Allowed values: Standard.")
         c.argument('tags', tags_type)
 
+    with self.argument_context('ppg create', min_api='2021-11-01') as c:
+        c.argument('zone', zone_type)
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/vm/_params.py:1416]: Same as https://github.com/Azure/azure-cli/pull/23167#discussion_r917722814
```suggestion
    with self.argument_context('ppg create') as c:
        c.argument('zone', zone_type, min_api='2021-11-01')
```

File: src/azure-cli/azure/cli/command_modules/vm/_validators.py [hunk 9]
```diff
@@ -2129,6 +2129,17 @@ def _disk_encryption_set_format(cmd, namespace, name):
...
 
 
+def process_ppg_create_namespace(namespace):
+    """
+    The availability zone can be provided only when an intent is provided
+    """
+    if namespace.zone and not namespace.intent_vm_sizes:
+        raise RequiredArgumentMissingError('The availability zone can be provided only when an intent is provided. '
+                                           'Please use parameter --intent-vm-sizes to specify possible sizes of '
+                                           'virtual machines that can be created in the proximity placement group.')
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/vm/_validators.py:2139]:
```suggestion
        raise RequiredArgumentMissingError('The --zone can be provided only when an intent is provided. '
                                           'Please use parameter --intent-vm-sizes to specify possible sizes of '
                                           'virtual machines that can be created in the proximity placement group.')
```

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**